pkg/controller/podautoscaler/replica_calculator.go

```diff
@@ -340,13 +349,26 @@ func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(currentReplicas int3
 	return replicaCount, utilization, timestamp, nil
 }
 
-func groupPods(pods []*v1.Pod, metrics metricsclient.PodMetricsInfo, resource v1.ResourceName, cpuInitializationPeriod, delayOfInitialReadinessStatus time.Duration) (readyPodCount int, ignoredPods sets.String, missingPods sets.String) {
+func groupPods(pods []*v1.Pod, metrics metricsclient.PodMetricsInfo, resource v1.ResourceName, cpuInitializationPeriod, delayOfInitialReadinessStatus time.Duration) (readyPodCount int, ignoredPods sets.String, missingPods sets.String, hasMultipleOwners bool) {
 	missingPods = sets.NewString()
 	ignoredPods = sets.NewString()
+	hasMultipleOwners = false
+	var podOwnerID types.UID
 	for _, pod := range pods {
 		if pod.DeletionTimestamp != nil || pod.Status.Phase == v1.PodFailed {
+			// remove pod from metrics
+			delete(metrics, pod.Name)
 			continue
 		}
+		if len(pod.OwnerReferences) > 0 {
+			if podOwnerID != pod.OwnerReferences[0].UID {
+				if podOwnerID != types.UID("") {
```

Comment:
why we need this check?

Reply:
Sorry for the late response, i was on vacation.
I am not sure whether you were asking about the if statements or about why I want to check whether pods have multiple owners.
For those if statements, I want to check whether 'podOwnerID' is assigned a valid uid yet. (by default, its avlue is ""), if it was already assigned a valid owner uid, but is different from current pod's owner uid, then we can conclude that the pods have multiple owners.

I think counting owners is one solution to distinguish those situations, e.g. rolling upgrade,  that I do not want hpa to take actions.

Reply:
Thanks for the clarification, you cleared my doubt.